src/pages/index.js

```diff
@@ -91,53 +75,106 @@ function renderCard(data) {
 const imagePopup = new PopupWithImage("#preview-image-modal");
 imagePopup.setEventListeners();
 
-//FORM POPUP
+//DELETE CARD POPUP
+const deleteCardPopup = new PopupWithConfirmation("#card__delete-modal");
+deleteCardPopup.setEventListeners();
+
+const profileUpdateSubmit = new PopupWithConfirmation("#profile-edit-modal");
+const newCardSubmit = new PopupWithConfirmation("#add-new-card");
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setSubmitText(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteCardPopup.close();
+        card.removeCard();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => deleteCardPopup.setSubmitText(false));
+  });
+}
 
+//FORM POPUP
 const openImagePopup = new PopupWithForm(
   "#add-new-card",
   handleImageFormSubmit
 );
-const editProfile = new PopupWithForm(
-  "#profile-edit-modal",
-  handleFormSubmit
+const editProfile = new PopupWithForm("#profile-edit-modal", handleFormSubmit);
+const profilePicEdit = new PopupWithForm(
+  "#profile__image-edit",
+  profilePicEditSubmit
 );
+
 editProfile.setEventListeners();
+profilePicEdit.setEventListeners();
 openImagePopup.setEventListeners();
 
 const editFormValidator = new FormValidator(config, profileModalForm);
+const profilePicValidator = new FormValidator(config, profilePicModal);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```